pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -553,9 +553,8 @@ void collectStatsAndIndexCreationInfo()
       ColumnStatistics columnProfile = _segmentStats.getColumnProfileFor(column);
       DictionaryIndexConfig dictionaryIndexConfig = indexConfigsMap.get(column).getConfig(StandardIndexes.dictionary());
       boolean createDictionary = dictionaryIndexConfig.isDisabled();
-      boolean useVarLengthDictionary =
-          dictionaryIndexConfig.getUseVarLengthDictionary() || DictionaryIndexType.shouldUseVarLengthDictionary(
-              storedType, columnProfile);
+      boolean useVarLengthDictionary = dictionaryIndexConfig.getUseVarLengthDictionary()
+          || DictionaryIndexType.optimizeTypeShouldUseVarLengthDictionary(storedType, columnProfile);
```

Comment:
will this change backward compatibility? A new check `columnStoredType == FieldSpec.DataType.STRING` is being added?

Reply:
Good question - it won't, since this value `useVarLengthDictionary` is currently set but unused. Only when the new config is set to true, it will be used (false by default)

Reply:
thanks for sharing the context. stamped.